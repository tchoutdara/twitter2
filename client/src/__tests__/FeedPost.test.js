import React from 'react';
import { shallow, mount } from 'enzyme';
import ConnectedComponent, { FeedPost } from '../components/FeedPost';
import toJson from 'enzyme-to-json';
import { reduxHelper } from './reduxHelper';
const initialState = { posts: [] }

describe('<FeedPost />', () => {

    describe('render', () => {
      let component;
      beforeEach( () => {
        const tree = reduxHelper(ConnectedComponent, initialState).component
        component = mount(tree);
      });
  
      it ('matches snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
      });
  
    });

    it('updates state on change', () => {
        let component = shallow(<FeedPost posts={[]}/>)
        let input = component.find('input');
        input.simulate('focus');
        input.simulate('change', { target: { name: 'name', value: 'Hello' } })
        expect(component.state('name')).toEqual('Hello');
      });
     
      it('submits the form', () => {
        let test = { dispatch: jest.fn() }
        let component = mount(<FeedPost dispatch={test.dispatch} posts={[]}/>)
        component.setState({ name: 'Hello' });
        component.find('form').simulate('submit');
        expect(test.dispatch).toHaveBeenCalledTimes(1);
        expect(component.state('name')).toEqual('');
      });
});