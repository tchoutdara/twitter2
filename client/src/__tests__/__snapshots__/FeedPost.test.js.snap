// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedPost /> render matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(FeedPost)>
    <FeedPost
      dispatch={[Function]}
      user={
        Object {
          "posts": Array [],
        }
      }
    >
      <Segment>
        <div
          className="ui segment"
        >
          <Input
            action={true}
            fluid="true"
            onChange={[Function]}
            placeholder="Write something no one will care about..."
            type="text"
          >
            <div
              className="ui fluid action input"
            >
              <input
                key=".0"
                onChange={[Function]}
                placeholder="Write something no one will care about..."
                type="text"
              />
              <Button
                as="button"
                key=".1"
                onClick={[Function]}
              >
                <button
                  className="ui button"
                  onClick={[Function]}
                >
                  Post
                </button>
              </Button>
            </div>
          </Input>
        </div>
      </Segment>
    </FeedPost>
  </Connect(FeedPost)>
</Provider>
`;
